### DISCUSSION TITLE
"Unsupported Media Type" on create-user-request with Js-SDK

### DISCUSSION THREAD

**Discussion by iamgaddiel** [pocketbase/pocketbase#2538] [Q&A]
I`m trying to create a user by sending api request using Js-SDK

POST
 https://builders.pockethost.io/api/collections/users/records 

const data = { 
    "username": "test_username", 
    "email": "test@example.com", 
    "password": "12345678", 
    "passwordConfirm": "12345678", 
    "name": "test", 
    "skill": "test", 
     "phone": "test"
}; 

const record = await pb.collection('users').create(data);

and getting error:

{ 
    "errorDetails": "unsupported request content-type", 
    "errorMessage": "Failed to load the submitted data due to invalid formatting"
}

![Screenshot_20230522-193543_Firefox.png](https://github.com/pocketbase/pocketbase/assets/59672983/775d02ea-0a76-452a-866a-226d8d42437e)

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it locally.

The user agent seems strange. In what environment do you execute the request? ReactNative?

**Reply by iamgaddiel** [reply]
I did it in Ionic React, what's weird is it worked perfectly when I ran it locally but the error only occurs when I build and test it out in an apk file.

The User Agent is from a Samsung mobile phone.

**Reply by ganigeorgiev** [reply]
I'm not familiar with Ionic React and I haven't tested the JS SDK with it.

The JS SDK internally uses `fetch` for sending requests so I guess it could be something related to that if Ionic ships their own implementation, I'm not really sure.

Alternatively, you can try to load a polyfill like https://github.com/lquixada/cross-fetch and see if that will fix the issue.

**Reply by iamgaddiel** [reply]
Thanks for pointing me in the right direction, but I do want to ask, is there a way log server request sent to pocketbase/pockethost.io  aside from the log on admin panel?

**Reply by ganigeorgiev** [reply]
PocketBase doesn't logs the raw requests. I'm not sure about pockethost.io, you may want to ask in their repo.

I haven't tested it, but you can try to use the Chrome dev tools with Ionic to see what is the request data you are sending - https://ionicframework.com/docs/troubleshooting/debugging.

**Reply by iamgaddiel** [reply]
Thanks a lot for your help.
